sofdev6-02-2024/TheBoys
# feat: view a specific routine in detail

the view was made to be able to see the exercises of a routine and at the same time to be able to see the details of an exercise and to be able to change its repetitions and to change the state of the exercise.

To test it you must follow these steps: 

1. create a user for the moment with : POST - http://localhost:4444/users/
2. create routines, this can be done from the frond end.
3. manually you must relate the created routines with your newly created user in the database .

![Screenshot from 2024-11-20 10-24-33](https://github.com/user-attachments/assets/c962434a-264f-447f-9290-8ba19e0adf33)

5. in the RoutinesGrid classes and in the listRoutines page where is the line `http://localhost:4444/routines/user/${userId}` replace the userId with the one of your created user .
6. in the page class of the listRoutines , 
 in this part remove the validations of userID to know if the user is logged in : useEffect(() => {
    if (!id || !userId) {
      if (!id) setError(“A routine ID was not provided.”);
      if (!userId) setError(“User is not logged in.”);
      setLoading(false);
      return;
    }

7. finally you can test from the frond end the routines and see their exercises
